### PR TITLE
libdatrie 0.2.12 (new formula)

### PR DESCRIPTION
**Pull request by phondanai**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by DomT4** [comment]
> GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

Willing to ignore that because this is packaged by Debian.

File: Formula/libdatrie.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Libdatrie < Formula
+  desc "Double-Array Trie Library"
+  homepage "https://linux.thai.net/projects/datrie"
+  url "https://github.com/tlwg/libdatrie/releases/download/v0.2.12/libdatrie-0.2.12.tar.xz"
+  sha256 "452dcc4d3a96c01f80f7c291b42be11863cd1554ff78b93e110becce6e00b149"
+
+  depends_on "autoconf" => :build
```
Comment (DomT4) [inline Formula/libdatrie.rb]: It seems possible we might be able to remove all of these dependencies apart from  `doxygen` based on the release tarball, which we're using, having a pre-generated `configure` script already.

File: Formula/libdatrie.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--prefix=#{prefix}",
+                          "--enable-shared"
+    system "make", "install-am"
+    system "make", "install-exec"
+  end
+
+  test do
+    system "#{bin}/trietool"
```
Comment (DomT4) [inline Formula/libdatrie.rb]: Any chance of a test that exercises a deeper level of functionality? Thanks!

**Review by DomT4 — CHANGES_REQUESTED** [review]

File: Formula/libdatrie.rb (hunk 2 above)
Comment (phondanai) [reply Formula/libdatrie.rb]: Is it possible to run test cases (make check) inside `test do` as a result of the test functionality?
Thanks.

Comment (DomT4) [reply Formula/libdatrie.rb]: It won't work, I'm afraid. The `testpath` and `buildpath` are separate locations that don't exist beyond the point where `test` or `install` has finished executing.

**Review by javian — CHANGES_REQUESTED** [review]

**Pull request closed without merging**